src/dir.rs

```diff
@@ -225,15 +225,13 @@ impl Entry {
     pub fn ino(&self) -> u64 {
         cfg_if! {
             if #[cfg(any(target_os = "aix",
-                         target_os = "android",
                          target_os = "emscripten",
                          target_os = "fuchsia",
                          target_os = "haiku",
-                         target_os = "illumos",
+                         solarish,
+                         linux_android,
                          apple_targets,
-                         target_os = "l4re",
-                         target_os = "linux",
-                         target_os = "solaris"))] {
+                         target_os = "l4re"))] {
```

Comment:
Honestly we could probably eliminate l4re.  I'm not sure why it's there in the first place.  This is the only reference to it in Nix, and it doesn't come remotely close to compiling.

Reply:
Will do it